File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int start = 0, end = height.size()-1; 
        int max_ = 0; 
        while(start < end){
            int x = end - start; 
            int y = min(height[start],height[end]); 
            max_ = max(max_, x * y); 
            if(height[start] < height[end]){
                start++; 
            } else if(height[start] > height[end]){
                end--; 
            } else{
                start++;
                end--; 
            }
        }
        return max_; 
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length-1; 
        int answer = 0;
        while(left < right){

            int width = right - left; 
            answer = Math.max(answer, width * Math.min(height[left], height[right])); 

            if(height[left] > height[right]){
                right--; 
            } else{
                left++; 
            }
        }

        return answer; 
    }
}